pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/exchange/SingletonExchange.java

```diff
@@ -41,7 +42,13 @@ class SingletonExchange extends BlockExchange {
   protected void route(List<SendingMailbox> mailbox, TransferableBlock block)
       throws Exception {
     for (SendingMailbox sendingMailbox : mailbox) {
-      sendBlock(sendingMailbox, block);
+      if (isLocal(sendingMailbox)) {
```

Comment:
TODO for this PR
- Throw when 0, or more than 1 exchange is configured ( test on block exchange unit test)
- Do not allow this ok worker assignment strategy  in the first place

Reply:
done